build/common.targets

```diff
@@ -388,9 +388,25 @@ Condition=" ('%(Extension)' == '.dll' OR '%(Filename)' == 'NuGet.CommandLine.XPl
       Condition=" %(Reference.NuGetPackageId) == 'Newtonsoft.Json' AND %(Reference.NuGetPackageVersion) != '$(NewtonsoftJsonPackageVersion)' " />
   </Target>
 
-  <Target Name="EnsurePackageReferenceVersions">
+   <Target Name="EnsurePackageReferenceVersions">
+    <ItemGroup Condition=" '$(TargetFrameworks)' != '' ">
+      <_TargetFrameworkItems Include="$(TargetFrameworks.Split(';'))" />
+    </ItemGroup>
+    <ItemGroup Condition=" '$(TargetFrameworks)' == '' ">
+      <_TargetFrameworkItems Include="$(TargetFramework)" />
+    </ItemGroup>
+    <MSBuild
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="EnsurePackageReferenceVersionsForFramework"
+      Properties="TargetFramework=%(_TargetFrameworkItems.Identity);"
+    />
+  </Target>
+
+  <Target Name="EnsurePackageReferenceVersionsForFramework">
+    <Message Importance="High" Text="+++++++++++++++ '$(TargetFramework)'    '$(IsCore)'"  />
```

Comment:
I missed, I agree with @zivkan, this should be removed. 

One bit that could be useful is if we add the framework for which the version is specified *in* the error message. 
But then you'd have to account for legacy vs non legacy projects. 

In particular, you could check whether TargetFramework is non null and if it is, add it to the error message, but I really don't think it's necessary.

Reply:
I was too lazy to check earlier, but this comment made me do it.  Legacy csproj doesn't have either `TargetFramework` or `TargetFrameworks`.  Therefore, I think the changes in this PR break checking legacy csproj.

Reply:
Are you suggesting it's not running for legacy? 
That should be fixable. Just don't batch if TargetFramework(s) are not set.

Reply:
Thank you for letting me know, it slipped through  my fingers. Just removed message I used for debugging purpose. 
Also I made changes in `TestablePlugin.csproj` for setting versions in centralized place.

Reply:
@erdembayar Can you check whether the target runs for legacy package reference projects?

A good example is NuGet.Tools.

Reply:
It's skipped on NuGet.Tools.csproj for before and after my PR.
Before PR(dev):
![image](https://user-images.githubusercontent.com/8766776/128275086-36deecfb-d3fe-4dff-8f83-6fe769a37acd.png)

After with PR:
![image](https://user-images.githubusercontent.com/8766776/128274510-9d9d3117-4383-4409-a346-a9fb34f8c6ca.png)

Reply:
@nkolev92 
Do you have any other question? Would you like me create follow up issue for legacy projects?

Reply:
If I go to dev, add a version to a PackageReference in NuGet.Tools.csproj and run the following:

```
C:\Code\NuGet.Client [dev ≡ +0 ~1 -0 !]> msbuild /t:EnsurePackageReferenceVersionsInSolution .\build\build.proj
Microsoft (R) Build Engine version 17.0.0-preview-21329-01+1b7661f36 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 06-Aug-21 10:08:43 AM.
Project "C:\Code\NuGet.Client\build\build.proj" on node 1 (EnsurePackageReferenceVersionsInSolution target(s)).
Project "C:\Code\NuGet.Client\build\build.proj" (1) is building "C:\Code\NuGet.Client\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (60) on node 1 (EnsurePackageReferenceVersions target(s)).
C:\Code\NuGet.Client\build\common.targets(399,5): error : Microsoft.VisualStudio.Sdk should not have a version declared outside of packages.targets but found 1.0.0 [C:\Code\NuGet.Client\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj]
Done Building Project "C:\Code\NuGet.Client\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (EnsurePackageReferenceVersions target(s)) -- FAILED.

C:\Code\NuGet.Client\build\build.proj(320,7): error MSB4181: The "MsBuild" task returned false but did not log an error.
Done Building Project "C:\Code\NuGet.Client\build\build.proj" (EnsurePackageReferenceVersionsInSolution target(s)) -- FAILED.


Build FAILED.
``` 

I get a failure, so this suggests it's running on legacy PR projects. 

I think if we merge it as is, we'd be merging a regression.

Reply:
It looks I misunderstood your concern previously, I was testing without adding packagereference with explicit version. Now I get same error with my branch too if I specify exact version, so that means no regression.
![image](https://user-images.githubusercontent.com/8766776/128555781-ce68903a-d5e3-46f9-948b-f1db557544b4.png)


```
C:\NuGetProj\NuGet.Client3 [dev-eryondon-9790EnsurePackageReferenceVersions ≡ +0 ~1 -0 !]> msbuild /t:EnsurePackageReferenceVersionsInSolution .\build\build.proj
Microsoft (R) Build Engine version 17.0.0-preview-21329-01+1b7661f36 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 8/6/2021 11:24:26 AM.
Project "C:\NuGetProj\NuGet.Client3\build\build.proj" on node 1 (EnsurePackageReferenceVersionsInSolution target(s)).
Project "C:\NuGetProj\NuGet.Client3\build\build.proj" (1) is building "C:\NuGetProj\NuGet.Client3\src\NuGet.Clients\NuG
et.Tools\NuGet.Tools.csproj" (60) on node 1 (EnsurePackageReferenceVersions target(s)).
Project "C:\NuGetProj\NuGet.Client3\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (60) is building "C:\NuGetProj\Nu
Get.Client3\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (60:2) on node 1 (EnsurePackageReferenceVersionsForFramew
ork target(s)).
C:\NuGetProj\NuGet.Client3\build\common.targets(406,5): error : System.IO.FileSystem.Primitives should not have a versi
on declared outside of packages.targets but found '1.0.0' [C:\NuGetProj\NuGet.Client3\src\NuGet.Clients\NuGet.Tools\NuG
et.Tools.csproj]
Done Building Project "C:\NuGetProj\NuGet.Client3\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (EnsurePackageRefer
enceVersionsForFramework target(s)) -- FAILED.

Done Building Project "C:\NuGetProj\NuGet.Client3\src\NuGet.Clients\NuGet.Tools\NuGet.Tools.csproj" (EnsurePackageRefer
enceVersions target(s)) -- FAILED.

C:\NuGetProj\NuGet.Client3\build\build.proj(320,7): error MSB4181: The "MsBuild" task returned false but did not log an
 error.
Done Building Project "C:\NuGetProj\NuGet.Client3\build\build.proj" (EnsurePackageReferenceVersionsInSolution target(s)
) -- FAILED.


Build FAILED.
```
![image](https://user-images.githubusercontent.com/8766776/128558047-ac69ef90-e88c-4459-8f07-3e761dcf15d7.png)

Reply:
All good then. 
I have no concerns.